test/fakeintake/server/server.go

```diff
@@ -359,18 +354,18 @@ func (fi *Server) handleGetPayloads(w http.ResponseWriter, req *http.Request) {
 		}
 		jsonResp, err = json.Marshal(resp)
 	} else if serverstore.IsRouteHandled(route) {
-		payloads := fi.store.GetJSONPayloads(route)
-		// build response
+		payloads, payloadErr := fi.store.GetJSONPayloads(route)
+		if payloadErr != nil {
+			writeHTTPResponse(w, buildErrorResponse(payloadErr))
+			return
+		}
+
 		resp := api.APIFakeIntakePayloadsJsonGETResponse{
 			Payloads: payloads,
 		}
 		jsonResp, err = json.Marshal(resp)
 	} else {
-		writeHTTPResponse(w, httpResponse{
-			contentType: "text/plain",
-			statusCode:  http.StatusBadRequest,
-			body:        []byte("invalid route parameter"),
-		})
+		writeHTTPResponse(w, buildErrorResponse(fmt.Errorf("invalid route parameter")))
```

Comment:
note: move to json error response since in that case we know that the user is request JSON format in the response.